datarepo-clienttests/src/main/java/utils/KubernetesClientUtils.java

```diff
@@ -352,19 +350,40 @@ public static void changeReplicaSetSizeAndWait(int podCount) throws Exception {
 
   private static long printApiPodCount(V1Deployment deployment, String message)
       throws ApiException {
+    long apiPodCount = getApiPodCount(deployment);
+    logger.debug("Pod Count: {}; Message: {}", apiPodCount, message);
+    return apiPodCount;
+  }
+
+  private static long getApiPodCount(V1Deployment deployment) throws ApiException {
     String deploymentComponentLabel = deployment.getMetadata().getLabels().get(componentLabel);
+    // loop through the pods in the namespace
+    // find the ones that match the deployment component label (e.g. find all the API pods)
     long apiPodCount =
         listPods().stream()
             .filter(
                 pod ->
                     deploymentComponentLabel.equals(
                         pod.getMetadata().getLabels().get(componentLabel)))
             .count();
-    logger.debug("Pod Count: {}; Message: {}", apiPodCount, message);
     return apiPodCount;
   }
 
-  private static void printApiPods(V1Deployment deployment) throws ApiException {
+  private static long getApiPodAtStatusCount(V1Deployment deployment, String podStatus)
```

Comment:
Are the possible pod statuses listed in an enum anywhere?

Reply:
From looking around the kubernetes api, it doesn't look like there is an enum. I could create one?